vortex-array/src/compute/compare.rs

```diff
@@ -112,6 +112,14 @@ pub fn compare(
         vortex_bail!("Compare operations only support arrays of the same type");
     }
 
+    if left.dtype().is_struct() || right.dtype().is_struct() {
```

Comment:
you already checked that the types are equal on 111 so it's enough to check one of them?